app-editors/neovim/neovim-0.10.0-r1.ebuild

```diff
@@ -0,0 +1,129 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# RelWithDebInfo sets -Og -g
+CMAKE_BUILD_TYPE=Release
+LUA_COMPAT=( lua5-{1..2} luajit )
+inherit cmake lua-single optfeature xdg
+
+DESCRIPTION="Vim-fork focused on extensibility and agility"
+HOMEPAGE="https://neovim.io"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/neovim/neovim.git"
+else
+	SRC_URI="https://github.com/neovim/neovim/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="Apache-2.0 vim"
+SLOT="0"
+IUSE="+nvimpager test"
+
+# Upstream say the test library needs LuaJIT
+# https://github.com/neovim/neovim/blob/91109ffda23d0ce61cec245b1f4ffb99e7591b62/CMakeLists.txt#L377
+REQUIRED_USE="${LUA_REQUIRED_USE} test? ( lua_single_target_luajit )"
+# TODO: Get tests running
+RESTRICT="!test? ( test ) test"
+
+# Upstream build scripts invoke the Lua interpreter
+BDEPEND="${LUA_DEPS}
+	>=dev-util/gperf-3.1
+	>=sys-devel/gettext-0.20.1
+	virtual/libiconv
+	virtual/libintl
+	virtual/pkgconfig
+"
+# Check https://github.com/neovim/neovim/blob/master/third-party/CMakeLists.txt for
+# new dependency bounds and so on on bumps (obviously adjust for right branch/tag).
+# List of required tree-sitter parsers is taken from cmake.deps/deps.txt
+DEPEND="${LUA_DEPS}
+	>=dev-lua/luv-1.45.0[${LUA_SINGLE_USEDEP}]
+	$(lua_gen_cond_dep '
+		dev-lua/lpeg[${LUA_USEDEP}]
+		dev-lua/mpack[${LUA_USEDEP}]
+	')
+	$(lua_gen_cond_dep '
+		dev-lua/LuaBitOp[${LUA_USEDEP}]
+	' lua5-{1,2})
+	>=dev-libs/libuv-1.46.0:=
+	>=dev-libs/libvterm-0.3.3
+	>=dev-libs/msgpack-3.0.0:=
+	>=dev-libs/tree-sitter-0.20.9:=
+	>=dev-libs/tree-sitter-bash-0.21 <dev-libs/tree-sitter-bash-0.22
+	>=dev-libs/tree-sitter-c-0.21 <dev-libs/tree-sitter-c-0.22
+	>=dev-libs/tree-sitter-lua-0.1 <dev-libs/tree-sitter-lua-0.2
+	>=dev-libs/tree-sitter-markdown-0.2 <dev-libs/tree-sitter-markdown-0.3
+	>=dev-libs/tree-sitter-python-0.21 <dev-libs/tree-sitter-python-0.22
+	>=dev-libs/tree-sitter-query-0.3 <dev-libs/tree-sitter-query-0.4
+	>=dev-libs/tree-sitter-vim-0.4 <dev-libs/tree-sitter-vim-0.5
+	>=dev-libs/tree-sitter-vimdoc-2 <dev-libs/tree-sitter-vimdoc-3
```

Comment:
Maybe use `=cat/pkg-1.2*` style dependencies instead, given this explicitly pins to semver?